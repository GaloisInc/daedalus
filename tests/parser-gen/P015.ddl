{- Tests
   fun call
-}


def myfun (m: int) =
  (mygauss m) + 42

def mygauss (m : int) =
  if m == 0 then 0 else (m + mygauss (m - 1))

def A = { $$ = UInt8 }

def Main = { x = A, y = (^ myfun (x as int)) }
