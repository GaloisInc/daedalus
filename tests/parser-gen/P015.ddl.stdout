module P015
 
--- Imports:
 
--- Type defs:
type P015.Main = { x: uint 8
                 ; y: int
                 }
 
--- Rules:
 
rec value
  P015.mygauss (m : int) : int =
    if (m == 0) then 0 else m + (P015.mygauss (m - 1))
 
P015.myfun (m : int) : int =
  (P015.mygauss m) + 42
 
P015.A : Grammar (uint 8) =
  do ($$ : uint 8) <- GetByte
     pure $$
 
P015.Main : Grammar P015.Main =
  do (x : uint 8) <- P015.A
     (y : int) <- pure (P015.myfun (x as int))
     pure {x = x,
           y = y}
 
P015._A : Grammar {} =
  @GetByte
 
P015._Main : Grammar {} =
  P015._A
